staging/src/k8s.io/apiextensions-apiserver/pkg/cmd/server/testing/testserver.go

```diff
@@ -124,7 +124,11 @@ func StartTestServer(t Logger, _ *TestServerInstanceOptions, customFlags []strin
 	fs := pflag.NewFlagSet("test", pflag.PanicOnError)
 
 	featureGate := utilfeature.DefaultMutableFeatureGate
+
+	// Configure the effective version
 	effectiveVersion := utilversion.DefaultKubeEffectiveVersion()
+	effectiveVersion.SetEmulationVersion(featureGate.EmulationVersion())
+
```

Comment:
@Jefftree @aaron-prindle Does this look right?  With it in place I'm able to set emulation version and then set feature gates and retain tests that require feature disablement.

(cc @cici37 since this may be needed for post-GA strict cost feature testing)

Reply:
Oops, I forgot about https://github.com/kubernetes/kubernetes/pull/127643. I'll rebase this off of that PR.